store/src/main/java/org/apache/rocketmq/store/MessageExtEncoder.java

```diff
@@ -175,11 +176,11 @@ public PutMessageResult encodeWithoutProperties(MessageExtBrokerInner msgInner)
     public PutMessageResult encode(MessageExtBrokerInner msgInner) {
         this.byteBuf.clear();
 
-        if (CommitLog.isMultiDispatchMsg(messageStoreConfig, msgInner)) {
+        if (messageStoreConfig.isEnableLmq() && MixAll.topicAllowsLMQ(msgInner.getTopic())) {
```

Comment:
It should use `msgInner.needDispatchLMQ()` instead of `MixAll.topicAllowsLMQ(msgInner.getTopic())`, otherwise non-lmq messages will enter this branch and cause issues during writing.

Reply:
Exactly